### ISSUE TITLE
NumPy 2.0 - CI failures - AttributeError: `ptp`

### ISSUE THREAD

**Issue by jGaboardi** [pysal/esda#299]
* [CI (ubuntu-latest, ci/39.yaml)](https://github.com/pysal/esda/actions/runs/9847620870/job/27187986153#step:7:671)
* [CI (ubuntu-latest, ci/39-numba.yaml)](https://github.com/pysal/esda/actions/runs/9847620870/job/27187986312#step:7:867)

```
FAILED esda/tests/test_topo.py::TopoTester::test_isolation_options - AttributeError: `ptp` was removed from the ndarray class in NumPy 2.0. Use np.ptp(arr, ...) instead.
FAILED esda/tests/test_topo.py::TopoTester::test_isolation_valid - AttributeError: `ptp` was removed from the ndarray class in NumPy 2.0. Use np.ptp(arr, ...) instead.
FAILED esda/tests/test_topo.py::TopoTester::test_prominence_options - AttributeError: `ptp` was removed from the ndarray class in NumPy 2.0. Use np.ptp(arr, ...) instead.
FAILED esda/tests/test_topo.py::TopoTester::test_prominence_valid - AttributeError: `ptp` was removed from the ndarray class in NumPy 2.0. Use np.ptp(arr, ...) instead.
FAILED esda/tests/test_topo.py::TopoTester::test_to_elevation - AttributeError: `ptp` was removed from the ndarray class in NumPy 2.0. Use np.ptp(arr, ...) instead.
```

**Comment by jGaboardi** [comment]
So this was actually an issue over in `libpysal.cg.voronoi.voronoi_regions()` that was [fixed several weeks ago](https://github.com/pysal/libpysal/commit/a92290b588d5a9add9ae59ff8dd7a508d0ae739e#diff-aa904952b4065370c617c30b408683bc3603c706bca2afea2d63e14c63f220bcR102), but is failing here in the Python 3.9 testing because that is pulling [`libpysal==v4.8.0`](https://github.com/pysal/esda/actions/runs/9847620870/job/27187986153#step:5:179).

Seems the actual fix here is the update the CI testing matrix by:
* dropping 3.8 & 3.9
* adding 3.12

Thoughts?

**Comment by jGaboardi** [comment]
closed via #318
